cve_bin_tool/parsers/go.py

```diff
@@ -27,9 +78,23 @@ def run_checker(self, filename):
                     packages = False
                     continue
                 if packages:
-                    product = line.split(" ")[0].split("/")[-1]
-                    version = line.split(" ")[1][1:].split("-")[0].split("+")[0]
-                    vendor = self.find_vendor(product, version)
-                    if vendor is not None:
-                        yield from vendor
+                    parts = line.split(" ")
+                    if len(parts) >= 2:
+                        product = line.split(" ")[0].split("/")[-1]
+                        version = line.split(" ")[1][1:].split("-")[0].split("+")[0]
+                        vendors = self.find_vendor(product, version)
+                        if vendors is not None:
+                            for v in vendors:
+                                if self.is_valid_data(
```

Comment:
This is only for vendors, right? Because I don't think it would be beneficial to force purl generation if we don't have an appropriate product or version.

Reply:
I want to generate something regardless.  The use case for this is so if someone then generates an SBOM from our scan, we can give them something to edit manually to get better results.

But you're correct that you wouldn't try to search using something where we didn't know what the product was.  (version is more iffy -- we don't right now, but some similar programs would return all possible results or do something fancy with dates).